store/cache_test.go

```diff
@@ -151,8 +152,39 @@ func (s *cacheSuite) TestCleanup(c *C) {
 	c.Check(osutil.FileExists(filepath.Join(s.cm.CacheDir(), cacheKeys[1])), Equals, false)
 
 	// the newest files are still there
-	c.Check(osutil.FileExists(filepath.Join(s.cm.CacheDir(), cacheKeys[2])), Equals, true)
+	c.Check(osutil.FileExists(filepath.Join(s.cm.CacheDir(), cacheKeys[len(cacheKeys)-s.maxItems])), Equals, true)
 	c.Check(osutil.FileExists(filepath.Join(s.cm.CacheDir(), cacheKeys[len(cacheKeys)-1])), Equals, true)
+
+	c.Check(s.cm.Count(), Equals, s.maxItems)
+}
+
+func (s *cacheSuite) TestStats(c *C) {
+	_, testFiles := s.makeTestFiles(c, s.maxItems+12)
+	testFilesNames := map[string]bool{}
+	for _, tf := range testFiles {
+		testFilesNames[filepath.Base(tf)] = true
+	}
+
+	c.Check(s.cm.Count(), Equals, s.maxItems+12)
+
+	// Remove the test files again, they are now only in the cache, we should
+	// observe some candidates for pruning next.
+	for _, p := range testFiles {
+		err := os.Remove(p)
+		c.Assert(err, IsNil)
+	}
+	stats, err := s.cm.Stats()
+	c.Assert(err, IsNil)
+
+	c.Check(int(stats.TotalSize), Equals, 24)
+	c.Check(stats.TotalEntries, Equals, 17)
```

Comment:
If there are no PruneCandidates this would pass anyway, can we add a check? (and replace 17 while we're at it, so it's less of a magic number)
```suggestion
	c.Check(stats.TotalEntries, Equals, s.maxItems+12)
	c.Check(stats.PruneCandidates, HasLen, s.maxItems+12)
```